src/analyses/constant_propagator.cpp

```diff
@@ -589,11 +602,21 @@ bool constant_propagator_domaint::partial_evaluate_with_all_rounding_modes(
 }
 
 bool constant_propagator_domaint::replace_constants_and_simplify(
+  const valuest &known_values,
   exprt &expr,
-  const namespacet &ns) const
+  const namespacet &ns)
 {
-  bool did_not_change_anything = values.replace_const.replace(expr);
-  did_not_change_anything &= simplify(expr, ns);
+  bool did_not_change_anything = true;
+
+  while(!known_values.replace_const.replace(expr))
```

Comment:
While I think I understand the purpose of the while loop here, and the purpose of the final `if(...) simplify()` following the while loop, it's not entirely obvious so might be worth a comment or two to explain?

Reply:
Done.